zerver/views/auth.py

```diff
@@ -357,6 +357,20 @@ def get_dev_users(extra_users_count=10):
 
 def login_page(request, **kwargs):
     # type: (HttpRequest, **Any) -> HttpResponse
+
+    # If no authentication method is enabled, show error.
+    is_auth_possible = False
+    if password_auth_enabled():
+        is_auth_possible = True
+    if dev_auth_enabled():
+        is_auth_possible = True
+    if google_auth_enabled():
+        is_auth_possible = True
+    if ldap_auth_enabled():
+        is_auth_possible = True
+    if not is_auth_possible:
+        return HttpResponseRedirect('/no_auth_backend')
```

Comment:
Let's make this call a function in `zproject/backends.py` to get the truth value of whether auth is enabled.  That way, all the auth management code is in one place.  Same with the `accounts_home` logic below.

I think in either case we should be using the following for the check:

auth_enabled_helper(list(AUTH_BACKEND_NAME_MAP.keys()), realm)